dev-requirements.txt

```diff
@@ -7,6 +7,7 @@ python-dotenv
 pyzotero
 requests
 sentence_transformers
+voyageai
```

Comment:
Can you add an `extra` for this in `pyproject.toml`? That way we can do `pip install paper-qa[voyage-ai]`

Reply:
For posterity-- this is because the setuptools is configured as being "dynamic" which means it comes from an external file. Seems like setuptools doesn't support using both a dynamic file and having hard-coded dependencies within the pyproject.toml. 

I didn't want to have to build a separate external file just for this lib, so I kept it like this.

Reply:
It's really too bad. I made https://github.com/pypa/setuptools/issues/4335 for this.

I guess a fix would be to move the `dev-requirements.txt` into this `pyproject.toml`